### PR TITLE
🎨 UI improvements

### PR DESCRIPTION
**Pull request by yanivshaked**
Sorry for the multiple pull-requests (I'm trying to push the wechat assets picker to be integrated into MegaTok's next official version...)

This PR has the following improvements:
- Display localized folder label "Folder" (according to language) in the app bar path entity selector.
- Support for LTR/RTL language text direction (added to the AssetsPickerTextDelegate)
- Auto resize of box constraints for pathEntitySelector according to selected assets.

Some "after" screenshots:

Chinese app bar before selection:
image

Chinese app bar after selection:
image

Hebrew app bar before selection:
image

Hebrew app bar after selection:
image

### CODE REVIEW

File: lib/src/delegates/asset_picker_builder_delegate.dart [hunk 1]
```diff
@@ -775,7 +775,7 @@ class DefaultAssetPickerBuilderDelegate
           child: Text(
             provider.isSelectedNotEmpty && !isSingleAssetMode
                 ? '${Constants.textDelegate.confirm}'
-                    '(${provider.selectedAssets.length}/${provider.maxAssets})'
+                    ' (${provider.selectedAssets.length}/${provider.maxAssets})'
```
Comment (yanivshaked) [inline lib/src/delegates/asset_picker_builder_delegate.dart:778]: This space is required to make sure the parenthesis are not too close to the text

**Comment by AlexV525** [comment]
I personally don't agree with the default "Folder" implementation, since we followed the UI with the WeChat. Meanwhile you can implement the "Folder" part by extend your own delegate. So I think that part can be moved to a custom implementation. Then we can push other parts forward later.

**Comment by yanivshaked** [comment]
How about adding a parameter for UI falvor, or even using the SpecialPickerType for that, so we can have one flavor for WeChat and another flavor for MegaTok?

**Comment by AlexV525** [comment]
Sorry but it doesn't sounds good to me. At least you should implement a custom delegate and give it a shot.

**Comment by AlexV525** [comment]
Beside of this, other enhancements seems kind of valid on some point.

**Comment by yanivshaked** [comment]
Ok, I will try with the custom delegate and update.

**Comment by AlexV525** [comment]
(Tips: Extends `DefaultAssetPickerBuilderDelegate` instead of `AssetPickerBuilderDelegate`)

**Comment by yanivshaked** [comment]
> (Tips: Extends `DefaultAssetPickerBuilderDelegate` instead of `AssetPickerBuilderDelegate`)

@AlexV525  I'm trying to go in that direction. To achieve that I need to add a new parameters to the `pickAssets` method, that will allow me to overload the `DefaultAssetPickerProvider`. Is that what you meant?

**Comment by AlexV525** [comment]
No. Make your own delegate based on `DefaultAssetPickerProvider`, build your own pick method with the delegate you implemented.

**Comment by yanivshaked** [comment]
> No. Make your own delegate based on `DefaultAssetPickerProvider`, build your own pick method with the delegate you implemented.

I'm having trouble with that, can you please take a look at the changes?
https://github.com/12Tech12/flutter_wechat_assets_picker/tree/20210503_UI_Improvement

**Comment by AlexV525** [comment]
Is the problem still exist?

**Comment by yanivshaked** [comment]
I managed to solve it, but I need quite a lot of code duplication in order to add new text to the `AssetsPickerTextDelegate`, since it's impossible to extend a class with factory constructor in dart.
1. If it's ok, I will create a PR with Megatok's pattern as a new example.
2. You can review the changes there, maybe you have a better idea.
What do you think?

**Comment by AlexV525** [comment]
What's the intention that you need a new text in the text delegate?

**Comment by yanivshaked** [comment]
Adding the folder name in the upper bar (according to the language)

**Comment by AlexV525** [comment]
Why it needs to follow languages? It's a custom implementation for the MegaTok app as you mentioned before.

**Comment by yanivshaked** [comment]
I will try to explain. The way I see it there are two alternatives:
- I will fork from flutter_wechat_asset_picker and adjust it to create the MegaTok pattern as required.
- I will keep working with the official flutter_wechat_asset_picker and make the required changes externally in an example (correct use of parameters, custom picker, etc)

I prefer the 2nd option, since this way I can benifit from any work done on the official plugin without maintenance (merge upstream changes etc.), and the official plugin can also benefit from my updates (pushing more ideas into PRs).

**Comment by AlexV525** [comment]
I'll take a look with your branch if I got more time, before that I'm curious about your question:

1. The second alternative with a custom picker, by extending the default picker delegate, won't need much fields override as I thought. It'll only required you to super the arguments in the constructor.
2. Following languages is not the thing that the picker done currently. Instead we only provide some base fields for languages implementation to display. If you extends a custom picker delegate to build a "Folder" name slot, then the point is to make your slot follow languages, rather than the text delegate.

**Comment by yanivshaked** [comment]
1. Sorry but I don't understand what you mean. 
2. That is a possibility, but still then, if I overload `pickAssets` I need to use `SlidePageTransitionBuilder` to make a similar implementation (and for example, that is not public).

**Comment by yanivshaked** [comment]
In order to make your review easier, I have created a new branch with relevant changes on the lastest version:
https://github.com/12Tech12/flutter_wechat_assets_picker/tree/20210509_MegaTok_Pattern_Example

**Comment by AlexV525** [comment]
> That is a possibility

How can it be possible? Sorry I don't understand their relations here. I thought your app would be a fixed language, and change languages using inner setting with provided languages. Then you can use the i18n in your widget.

**Comment by AlexV525** [comment]
So I've checked your branch, and make some changes with comments. The `SlidePageTransitionBuilder` needs to be exported, which you mentioned before. Some other codes looks redundant to me.

**Comment by yanivshaked** [comment]
Thanks, I will review the changes.

**Comment by AlexV525** [comment]
In 5.4.1 it's been exported as `AssetPickerPageRoute`.

**Comment by yanivshaked** [comment]
fei chang gan xie!

**Comment by yanivshaked** [comment]
If I'm using my own localization infrastructure in the `pathEntitySelector`, then `HebrewMegatokAssetsPickerTextDelegate` is redundant, removing.

**Comment by yanivshaked** [comment]
After removing, it look much more reasonable.
Is it ok to push the MegaTok pattern example to the upstream?
It will be helpful since there are several enhancement I would like to suggest, and I can use the Megatok pattern example to work on them.

**Comment by AlexV525** [comment]
Sure. But I have to reorder the structure of the example, which will allow more custom delegates contributed if more people want to.

**Comment by AlexV525** [comment]
Check https://github.com/fluttercandies/flutter_wechat_assets_picker/blob/master/example/lib/customs/CONTRIBUTING.md for more details.

**Comment by yanivshaked** [comment]
Thanks!

**Pull request closed without merging**